cnf-certification-test/networking/services/services.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2020-2022 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package services
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/networking/netcommons"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func IsDualStack(aService *corev1.Service) (result bool, err error) {
```

Comment:
This functions seems a bit confusing to me. It supposed to tell you if the service is configured as dual stack (yes/no due to bool) but somehow there's a third valid possibility without being dual stack (single stack ipv6). Also, error messages related to problems getting the information you need (such as the version or the IP family policy) are mixed with error due to the service being not compliant.

Maybe an implementation that returns the type of the IP stack instead of a bool value would make it clearer. Just a suggestion, no need to change it if you don't see it that way.

Reply:
fixed